File: workspaces/dotnet/c-api1-core/src/GetVariant.cs
using System.Linq;
using OMP.LSWTSS.CApi1;

public static class GetVariant
{
    static Variant? _cachedVariant;

    public static Variant Execute()
    {
        if (_cachedVariant == null)
        {
            if (System.Environment.GetEnvironmentVariable("SteamAppId") != null)
            {
                _cachedVariant = Variant.Steam;
            }
            else if (System.Environment.GetCommandLineArgs().Any(x => x.Contains("-epicapp")))
            {
                _cachedVariant = Variant.EGS;
            }
            else
            {
                throw new System.InvalidOperationException();
            }
        }

        return _cachedVariant.Value;
    }
}

File: workspaces/rust/driver-library/src/get_variant.rs
use crate::*;

pub fn get_variant() -> Variant
{
    if std::env::var("SteamAppId").is_ok() {
        return Variant::KnownVariant(KnownVariant::Steam);
    }

    if std::env::args().any(|arg| arg.contains("-epicapp")) {
        return Variant::KnownVariant(KnownVariant::EGS);
    }

    return Variant::Unknown;
}
